src/iframe-entry/components/FeeSelect/FeeSelect.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+    Box,
+    BoxProps,
+    FeeOption,
+    List,
+    Select,
+    Selected,
+    Text,
+} from '@waves.exchange/react-uikit';
+import { TLong, TTransactionParamWithType } from '@waves/signer';
+import React, { FC, ReactElement, useCallback, useState } from 'react';
+import { WAVES } from '../../constants';
+import { IMeta } from '../../services/transactionsService';
+import { assetPropFactory } from '../../utils/assetPropFactory';
+import { getCoins, getPrintableNumber } from '../../utils/math';
+import { checkIsEnoughBalance } from './helpers';
+
+export type FeeSelectHandler = (fee: string, feeAssetId: string) => void;
+
+type Props = {
+    txMeta: IMeta<TTransactionParamWithType>;
+    fee: TLong;
+    onFeeSelect: FeeSelectHandler;
+    feeAssetId?: string | null;
+    availableBalance: TLong;
+};
+
+export const FeeSelect: FC<Props & BoxProps> = ({
+    txMeta,
+    fee: txFee,
+    feeAssetId,
+    onFeeSelect,
+    availableBalance,
+    ...rest
+}) => {
+    const getAssetProp = assetPropFactory(txMeta.assets);
+    const feeAsset = txMeta.assets[feeAssetId || ''] || WAVES;
+
+    const fee = getPrintableNumber(txFee, feeAsset.decimals);
+
+    const defaultFee: FeeOption = {
+        id: WAVES.assetId,
+        name: WAVES.name,
+        ticker: WAVES.ticker,
+        value: fee,
+    };
+
+    let feeList = txMeta.feeList.map((f) => ({
+        name: getAssetProp(f.feeAssetId, 'name'),
+        id: String(f.feeAssetId),
+        ticker: '',
+        value: String(f.feeAmount),
+    }));
+
+    const isEnoughBalance = checkIsEnoughBalance(
+        availableBalance,
+        txFee,
+        feeAsset.decimals
+    );
+    const hasNonDefaultFees = feeList.length;
+
+    if (hasNonDefaultFees) {
+        feeList = isEnoughBalance ? [defaultFee].concat(feeList) : feeList;
+    } else {
+        feeList = [defaultFee];
+    }
+
+    const [selectedFee, setSelectedFee] = useState<FeeOption>(feeList[0]);
+
+    const handleFeeSelect = useCallback(
+        (feeOption: FeeOption) => {
+            setSelectedFee(feeOption);
+
+            const feeAssetId = feeOption.id;
+            const fee = getCoins(
+                feeOption.value,
+                getAssetProp(feeOption.id, 'decimals')
+            );
+
+            onFeeSelect(fee, feeAssetId);
+        },
+        [getAssetProp, onFeeSelect]
+    );
+
+    return (
+        <Box {...rest}>
+            <Text variant="body2" color="basic.$500" display="block" mb="$5">
+                Fee
+            </Text>
+            {feeList.length > 1 ? (
```

Comment:
can we use `hasNonDefaultFees` here?

Reply:
we cant, cos `hasNonDefaultFees` is feeList.length > 0